lisa/sut_orchestrator/azure/platform_.py

```diff
@@ -1477,6 +1493,12 @@ def _node_generate_min_capability(
 
         # Generate min capability node
         min_cap: schema.NodeSpace = req_cap.generate_min_capability(azure_cap_copy)
+        assert isinstance(min_cap.data_disk_count, int)
```

Comment:
This logic should be above the `req_cap.generate_min_capability(azure_cap_copy)`, so the input may be IntRange. If not, when the iops is 1, the min size will be 1 above, and this logic doesn't know it needs an exact 1, min 1, or max 1. And it needs to more complex logic to check which iops data in the range in data disk's get min size.